src/overlays/actors/ovl_En_Heishi2/z_en_heishi2.c

```diff
@@ -21,77 +67,880 @@ const ActorInit En_Heishi2_InitVars = {
+            this->actionFunc = func_80A535BC;
+        }
+    }
+}
+
+void func_80A535BC(EnHeishi2* this, GlobalContext* globalCtx) {
+    f32 frames = SkelAnime_GetFrameCount(&D_06005500.genericHeader);
+    this->unk_2EC = frames;
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005500, 1.0f, 0.0f, frames, 2, -10.0f);
+    this->actionFunc = func_80A53638;
+}
+
+void func_80A53638(EnHeishi2* this, GlobalContext* globalCtx) {
+
+    Actor* thisx;
+    f32 frameCount;
+    BgSpot15Saku* gate;
+
+    frameCount = this->skelAnime.animCurrentFrame;
+    thisx = &this->actor;
+    gate = (BgSpot15Saku*)(globalCtx->actorCtx.actorList[7].first);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((frameCount >= 12.0f) && (!this->audioFlag)) {
+        Audio_PlayActorSound2(thisx, NA_SE_EV_SPEAR_HIT);
+        this->audioFlag = 1;
+    }
+    if (this->unk_2EC <= frameCount) {
+        if (gate != NULL) {
+
+            do {
+                if (ACTOR_BG_SPOT15_SAKU != gate->dyna.actor.id) {
+                    gate = (BgSpot15Saku*)(gate->dyna.actor.next);
+                }
+
+                else {
+                    this->attached = gate;
+                    gate->unk_168 = 1;
+                    break;
+                }
+            } while (gate != NULL);
+        }
+        // "I've come!"
+        osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆ きたきたきたぁ！ ☆☆☆ %x\n" VT_RST, gate->dyna.actor.next);
+        this->actionFunc = func_80A5372C;
+    }
+}
+
+void func_80A5372C(EnHeishi2* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005C30, 1.0f, 0.0f,
+                         (s16)(f32)SkelAnime_GetFrameCount(&D_06005C30.genericHeader), 0, -10.0f);
+    this->gateTimer = 200;
+    this->cameraId = Gameplay_CreateSubCamera(globalCtx);
+    Gameplay_ChangeCameraStatus(globalCtx, 0, 1);
+    Gameplay_ChangeCameraStatus(globalCtx, this->cameraId, 7);
+    VEC_SET(this->unk_280, 947.0f, 1195.0f, 2682.0f);
+    VEC_SET(this->unk_28C, 1164.0f, 1145.0f, 3014.0f);
+    func_800C04D8(globalCtx, this->cameraId, &this->unk_280, &this->unk_28C);
+    this->actionFunc = func_80A53850;
+}
+
+void func_80A53850(EnHeishi2* this, GlobalContext* globalCtx) {
+    BgSpot15Saku* gate;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_800C04D8(globalCtx, this->cameraId, &this->unk_280, &this->unk_28C);
+    gate = (BgSpot15Saku*)this->attached;
+    if ((this->gateTimer == 0) || (gate->unk_168 == 0)) {
+        Gameplay_ClearCamera(globalCtx, this->cameraId);
+        Gameplay_ChangeCameraStatus(globalCtx, 0, 7);
+        func_80106CCC(globalCtx);
+        this->unk_30C = 1;
+        func_8002DF54(globalCtx, NULL, 7);
+        this->actionFunc = func_80A531E4;
+    }
+}
+
+void func_80A53908(EnHeishi2* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06005C30.genericHeader);
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005C30, 1.0f, 0.0f, (s16)frameCount, 0, -10.0f);
```

Comment:
cast to s16 shouldnt be needed

Reply:
The cast is needed, else like half of the function shows up as red and missing. I dont know whats up with the SkelAnime_ChangeAnim function, but I've never gotten anything to match with it without casting first.

Reply:
I think it might have to do with SkelAnime_GetFrameCount